### PR TITLE
Added MSN3700/MSN3700C platform config

### PR DESCRIPTION
**Pull request by nazariig**
**- What I did**
Added MSN3700/MSN3700C platform config

**- How I did it**
* N/A

**- How to verify it**
* N/A

**- Description for the changelog**
* N/A

### CODE REVIEW

File: device/mellanox/x86_64-mlnx_msn3700-r0/ACS-MSN3700/buffers.json.j2 [hunk 1]
```diff
@@ -0,0 +1 @@
+../../x86_64-mlnx_msn2700-r0/ACS-MSN2700/buffers.json.j2
```
Comment (liat-grozovik) [inline device/mellanox/x86_64-mlnx_msn3700-r0/ACS-MSN3700/buffers.json.j2]: not sure i understand what you did
you created a new file for buffers for sn3700 but you are using sn2700 buffers
according to the excel the buffer for sn3700 are not the same for sn2700

File: device/mellanox/x86_64-mlnx_msn3700-r0/ACS-MSN3700/port_config.ini [hunk 2]
```diff
@@ -0,0 +1 @@
+../../x86_64-mlnx_msn2700-r0/ACS-MSN2700/port_config.ini
```
Comment (liat-grozovik) [inline device/mellanox/x86_64-mlnx_msn3700-r0/ACS-MSN3700/port_config.ini]: what am i missing? i expect to see new independed port_config.ini for sn3700 not the one used for sn2700

File: files/build_templates/buffers_config.j2 [hunk 3]
```diff
@@ -21,6 +21,10 @@ def
 {%-     set switch_role      = '' %}
 {%- endif -%}
 
+{%- if DEVICE_NEIGHBOR_METADATA is not defined %}
```
Comment (liat-grozovik) [inline files/build_templates/buffers_config.j2]: how come this change related to sn3700 support?
if this is a required fix it should be regardless of sn3700 PR

File: platform/mellanox/mlnx-fw-upgrade.j2 [hunk 4]
```diff
@@ -4,39 +4,65 @@ declare -r SCRIPT_NAME="$(basename "$0")"
...
+declare -r VERBOSE_WARNING="2"
+declare -r VERBOSE_NOTICE="3"
+declare -r VERBOSE_INFO="4"
+
+declare -r VERBOSE_MAX="${VERBOSE_INFO}"
+declare -r VERBOSE_MIN="${VERBOSE_ERROR}"
+
 declare -r EXIT_SUCCESS="0"
-declare -r EXIT_ERROR="1"
+declare -r EXIT_FAILURE="1"
```
Comment (liat-grozovik) [inline platform/mellanox/mlnx-fw-upgrade.j2]: why failure and not error? what is the idea behind the name change?

File: platform/mellanox/mlnx-fw-upgrade.j2 [hunk 5]
```diff
@@ -4,39 +4,65 @@ declare -r SCRIPT_NAME="$(basename "$0")"
...
+IMAGE_UPGRADE="${NO_PARAM}"
+VERBOSE_LEVEL="${VERBOSE_MIN}"
 
 function PrintHelp() {
     echo
     echo "Usage: ./${SCRIPT_NAME} [OPTIONS]"
     echo
     echo "OPTIONS:"
-    echo "  -u, --upgrade  Upgrade MLNX ASIC firmware using next boot image (useful after SONiC-To-SONiC update)"
+    echo "  -u, --upgrade  Upgrade ASIC firmware using next boot image (useful after SONiC-To-SONiC update)"
```
Comment (liat-grozovik) [inline platform/mellanox/mlnx-fw-upgrade.j2]: as this is MLNX ASIC, any reason why taking MLNX out of it? In any case suggest to use Mellanox and not MLNXS

File: files/build_templates/buffers_config.j2 (hunk 3 above)
Comment (mykolaf) [inline files/build_templates/buffers_config.j2]: Looks like a WA for some issue which affects all platforms. Consider having it as a separate PR?

File: platform/mellanox/mlnx-fw-upgrade.j2 [hunk 6]
```diff
@@ -47,65 +73,138 @@ function ParseArguments() {
...
-        exit "${EXIT_ERROR}"
+        ExitFailure "could not retreive available FW version"
     fi
 
     if [[ "${_FW_CURRENT}" == "${_FW_AVAILABLE}" ]]; then
-        echo "Mellanox firmware is up to date"
+        ExitSuccess "firmware is up to date"
     else
-        echo "Mellanox firmware upgrade is required. Installing compatible version..."
+        LogNotice "firmware upgrade is required. Installing compatible version..."
```
Comment (liat-grozovik) [inline platform/mellanox/mlnx-fw-upgrade.j2]: why removing Mellnaox?

File: device/mellanox/x86_64-mlnx_msn3700-r0/ACS-MSN3700/buffers.json.j2 (hunk 1 above)
Comment (nazariig) [reply device/mellanox/x86_64-mlnx_msn3700-r0/ACS-MSN3700/buffers.json.j2]:
```suggestion
../../x86_64-mlnx_msn2700-r0/ACS-MSN2700/buffers.json.j2
```
This is a template which are common for all our platforms.
It does not contain any config itself:
{%- set default_topo = 't0' %}
{%- include 'buffers_config.j2' %}

File: device/mellanox/x86_64-mlnx_msn3700-r0/ACS-MSN3700/port_config.ini (hunk 2 above)
Comment (nazariig) [reply device/mellanox/x86_64-mlnx_msn3700-r0/ACS-MSN3700/port_config.ini]: @liat-grozovik 
This file is the same for both MSN2700 and MSN3700 (both has the same amount of interfaces) thanks to the SAI team, who provided a unified logic for interfaces port mapping (SAI HW lanes and HW Spec lanes is not the same).

File: files/build_templates/buffers_config.j2 (hunk 3 above)
Comment (nazariig) [reply files/build_templates/buffers_config.j2]: @liat-grozovik this should be a fix, but for now we are using it as WA.

File: platform/mellanox/mlnx-fw-upgrade.j2 (hunk 4 above)
Comment (nazariig) [reply platform/mellanox/mlnx-fw-upgrade.j2]: @liat-grozovik just a name convention. Nothing special.

File: files/build_templates/buffers_config.j2 (hunk 3 above)
Comment (liat-grozovik) [reply files/build_templates/buffers_config.j2]: in this case it should not go to the formal PR

File: platform/mellanox/mlnx-fw-upgrade.j2 (hunk 4 above)
Comment (liat-grozovik) [reply platform/mellanox/mlnx-fw-upgrade.j2]: i think we should continue with error. this is more convention indication for error

**Pull request closed without merging**